### PR TITLE
LPS-144624 Adds Clay's new TreeView component to SelectCategory

### PR DESCRIPTION
**Pull request by matuzalemsteles**
Hey @liferay-echo team,

This PR implements the new TreeView component from Clay to SelectCategory, so it has some visual changes to adopt the new Lexicon look. I'd like to get some feedback regarding this implementation and if I'm missing something.

- Multiple selection is done with Checkbox
- Single selection is not using the Checkbox but the selection in Node
- The filter preserves tree structure instead of a flat list

The tests I did were adding categories and testing on the blog, let me know if there are other places I'm missing, thanks guys! Any suggestion is welcome.

### CODE REVIEW

File: modules/apps/asset/asset-categories-selector-web/src/main/resources/META-INF/resources/js/SelectCategory.es.js [hunk 1]
```diff
@@ -50,66 +87,84 @@ function SelectCategory({
...
+		// Traverse the tree only once to get the pre-selected items. It is cheaper
+		// to control the unchecked state by iterating over the array instead of
+		// traversing the tree each change.
 
-		const openerWindow = Liferay.Util.getOpener();
+		if (!isComputePreSelectedRef.current) {
+			selectedItemsRef.current = {
+				...selectedItemsRef.current,
+				...computePreSelectedItems(items),
+			};
```
Comment (matuzalemsteles) [inline modules/apps/asset/asset-categories-selector-web/src/main/resources/META-INF/resources/js/SelectCategory.es.js:133]: Ideally, it would be nice if we can avoid this but as we need the data of the initially selected items I tried to make it only need it once, I'm also thinking how we can help with this on the Clay side.

File: modules/apps/asset/asset-categories-selector-web/package.json [hunk 2]
```diff
@@ -1,5 +1,7 @@
 {
 	"dependencies": {
+		"@clayui/core": "3.46.0",
```
Comment (matuzalemsteles) [inline modules/apps/asset/asset-categories-selector-web/package.json:3]: This new version came out today, this PR depends on it, so we would have to wait for the PR https://github.com/liferay-frontend/liferay-portal/pull/1903 enter first and then send this one.

**Comment by julien** [comment]
Hey @matuzalemsteles sorry I didn't mention this, but I didn't know you where going to send the pull request.
We first need to add a feature flag for this feature since the echo team asked us for this.
I'll have a look at that today, and get back to you once it's done.

**Comment by matuzalemsteles** [comment]
No problem, thanks @julien for the info!

**Comment by julien** [comment]
@matuzalemsteles the feature flag "functionality" is in this commit https://github.com/liferay-echo/liferay-portal/pull/7185/commits/a73b936f8b27b33e7e750cb23ba86e76d0f06c20 (if you need help understanding how it works, please tell me) - To use it from JavaScript you can have a look at this https://github.com/liferay-echo/liferay-portal/pull/7185/files#diff-7e088f43160568b645342632b867ee0ef46760c7d5f651259db4ad05317457d6R79

**Comment by matuzalemsteles** [comment]
Okay, I redid the changes to now use the Feature Flag that entered the master. I think this is ready for review.

File: modules/apps/asset/asset-categories-selector-web/src/main/resources/META-INF/resources/js/SelectTree.es.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {TreeView} from '@clayui/core';
```
Comment (julien) [inline modules/apps/asset/asset-categories-selector-web/src/main/resources/META-INF/resources/js/SelectTree.es.js:15]: @matuzalemsteles  not a blocker but just so you know, up to now when migrating to the new `TreeView` component, we've used 

```js
import {TreeView as ClayTreeView} from '@clayui/core';
```

The main reason is that product teams are always seen the "Clay" prefix in our "component names". As I said, it's not a blocker, so we can do the following:

1. Keep it as it is in this change if our colleagues from the Echo team agree
2. Export the component as `ClayTreeView` directly in Clay like we do for almost all of our components
3. Use `import {TreeView as ClayTreeView}`

Comment (matuzalemsteles) [reply modules/apps/asset/asset-categories-selector-web/src/main/resources/META-INF/resources/js/SelectTree.es.js:15]: Hmm this is interesting, thanks for the information Julien. I'll keep `import {TreeView as ClayTreeView}` for now, I think it's worth opening an issue to find out if we should go with the Clay prefix in the component names for the `@clayui/core` package and see what the people think. I'm fine with any decision.

Comment (julien) [reply modules/apps/asset/asset-categories-selector-web/src/main/resources/META-INF/resources/js/SelectTree.es.js:15]: @matuzalemsteles yes I agree - on the Clay side, I don't mind keeping it as is. It's mostly for product teams that this comes from Clay, even though it seems obvious to me. I had to explain it [this](https://github.com/liferay-lima/liferay-portal/pull/2715#issuecomment-1031660943) thread (it might hard to follow because of GitHub's UI), but my conclusion is that this is not a blocker.

**Comment by victorg1991** [comment]
Thanks @matuzalemsteles I'll review this tomorrow :)

**Comment by victorg1991** [comment]
Resent here: https://github.com/liferay-echo/liferay-portal/pull/7247 Thanks @matuzalemsteles and @julien 😍
